core/src/main/java/smile/regression/OLS.java

```diff
@@ -22,6 +22,7 @@
 import org.slf4j.LoggerFactory;
 import smile.math.Math;
 import smile.math.matrix.Matrix;
+import smile.math.matrix.JMatrix;
```

Comment:
JMatrix should never be directly used. Instead, you should declare a matrix as DenseMatrix and use Matrix.newInstance() to create one. In this way, the system will automatically pick up a native implementation if it is available. If a native implementation is not available, it will fall back to JMatrix.